src/ngraph/pass/algebraic_simplification.hpp

```diff
@@ -29,10 +30,12 @@ namespace ngraph
 class ngraph::pass::AlgebraicSimplification : public FunctionPass
 {
 public:
-    AlgebraicSimplification()
+    AlgebraicSimplification(pass::PassPropertyMask property =
```

Comment:
The pass properties are intrinsic to the pass or matcher. So it should only be set by the pass or matcher itself not passed in from constructor. I also made `set_property()` protected so that it can't be called externally.

Reply:
Sorry, to correct what I said, the `Matcher` class is a bit different than the pass. The Matcher constructor does allow the property to be passed in during construction. The reason is the callback function needs to have these pass properties. I'm thinking maybe it's better to make the pass property a parameter of the callback, I'll fix that in another PR.